### PR TITLE
feat: sample unification

### PR DESCRIPTION
**Pull request by uoboda-splunk**
With this PR we introduce:

- sample unification of requirement tests and KO tests samples
- refactor requirement tests, which are now part of KO tests
- requirement unit tests are now part of PSA, previously it was separate project
- update in unit and e2e tests

It will be released as a BREAKING CHANGE

### CODE REVIEW

File: .github/workflows/build-test-release.yml [hunk 1]
```diff
@@ -198,7 +199,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        splunk-version: [8.1, 8.2]
+#        splunk-version: [8.1, 8.2]
+        splunk-version: [8.2]
```
Comment (kkania-splunk) [inline .github/workflows/build-test-release.yml:203]: probably we should test with 3 last splunk versions
```suggestion
        splunk-version: [8.1, 8.2, 9.0]
```

File: pytest_splunk_addon/standard_lib/event_ingestors/ingestor_helper.py [hunk 2]
```diff
@@ -104,18 +104,18 @@ def ingest_events(
...
+        #     )
+        #     for input_type, events in requirement_events_dict.items():
+        #         LOGGER.debug(
+        #             "Received the following input type for HEC event: {}".format(
+        #                 input_type
+        #             )
+        #         )
+        #         event_ingestor = cls.get_event_ingestor(input_type, ingest_meta_data)
+        #         event_ingestor.ingest(events, thread_count)
+        #     LOGGER.info("Ingestion Done")
```
Comment (kkania-splunk) [inline pytest_splunk_addon/standard_lib/event_ingestors/ingestor_helper.py:121]: code to be deleted
```suggestion
```

File: Dockerfile.uf [hunk 3]
```diff
@@ -13,10 +13,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-ARG SPLUNK_VERSION=latest
-FROM splunk/universalforwarder:$SPLUNK_VERSION
-ARG SPLUNK_VERSION=latest
+ARG SPLUNK_VERSION=8.2.6
+FROM splunk/universalforwarder:8.2.6
+ARG SPLUNK_VERSION=8.2.6
```
[the remark this answers is not in the document]
Comment (uoboda-splunk) [reply Dockerfile.uf:19]: Added comment to go back to latest when possible

**Review by kkania-splunk — APPROVED** [review]

**Pull request merged**